src/media/headset.ts

```diff
@@ -1,149 +1,201 @@
-    }
+import { Observable, Subject } from 'rxjs';
+import HeadsetService, { ConsumedHeadsetEvents, VendorImplementation } from 'softphone-vendor-headsets';
 
-    /**
-     * Gets the currently selected vendor implementation from the headset library
-     * @params none
-     * @returns VendorImplementation
-     */
-    get currentSelectedImplementation (): VendorImplementation {
-        return this.headsetLibrary.selectedImplementation;
-    }
-
-    /**
-     * Determines if the retry button is necessary to be rendered
-     * @params none
-     * @returns boolean
-     */
-    showRetry (): boolean {
-        const selectedImplementation = this.currentSelectedImplementation;
-        if (selectedImplementation?.disableRetry) {
-            return false;
-        }
-
-        return !!selectedImplementation
-            && !selectedImplementation.isConnected
-            && !selectedImplementation.isConnecting;
-    }
-
-    /**
-     * Attempts to reconnect to the selected vendor's SDK
-     * @param micLabel the label that matches the currently selected device
-     * @returns Promise<void>
-     */
-    async retryConnection (micLabel: string): Promise<void> {
-        return micLabel && this.headsetLibrary.retryConnection(micLabel);
-    }
-
-    /**
-     * Calls the headset library's incomingCall function to signal to the device
-     * to flash the answer call button's light to show an incoming call
-     * @param callInfo an object containing the conversationId and possible
-     * contactName for the incoming call that will be accepted or rejected
-     * @param hasOtherActiveCalls boolean determining if there are other active calls
-     * @returns Promise<void>
-     */
-    setRinging (callInfo: { conversationId: string, contactName?: string }, hasOtherActiveCalls): Promise<void> {
-        return this.headsetLibrary.incomingCall(callInfo, hasOtherActiveCalls);
-    }
-
-    /**
-     * Calls the headset library's outgoingCall function to signal to the device
-     * to switch on the answer call button's light to show an active call
-     * @param callInfo an object containing the conversationId and possible
-     * contactName for the call that is outgoing
-     * @returns Promise<void>
-     */
-    outgoingCall (callInfo: { conversationId: string, contactName: string }): Promise<void> {
-        return this.headsetLibrary.outgoingCall(callInfo);
-    }
+import GenesysCloudWebrtcSdk from '../client';
 
-    /**
-     * Calls the headset library's endCall function to signal to the device
-     * to switch off the answer call button's light to show the active call has ended
-     * @param conversationId a string representing the conversation that needs to be ended
-     * @returns Promise<void>
-     */
-    async endCurrentCall (conversationId: string): Promise<void> {
-        if (conversationId) {
-            return this.headsetLibrary.endCall(conversationId);
-        }
+export interface ISdkHeadset {
```

Comment:
That's fair. In that case it might be more beneficial to do an abstract class because then we wouldn't have to stub out the stub class, we could just extend the abstract class and be done (for the stub).